src/com/company/Main.java

```diff
@@ -36,145 +41,137 @@ public static void play() {
+//            for (PlayerSignType sign : signRow) {
+//                sign = PlayerSignType.EMPTY;
+//            }
+//        }
 
+
+        drawField(boardSize, board);
+        do {
             rowNumber = getRowNumber(boardSize);
             columnNumber = getColumnNumber(boardSize);
             //przenieść do funkcji załaduj do tablicy
-            System.out.println(CHOOSED + rowNumber + "/" + columnNumber);
-            if (tab[rowNumber - 1][columnNumber - 1] == 0) {
-                tab[rowNumber - 1][columnNumber - 1] = ItIsX;
+            System.out.println(SELECTED + rowNumber + "/" + columnNumber);
+            if (board[rowNumber - 1][columnNumber - 1] == PlayerSignType.EMPTY) {
+                board[rowNumber - 1][columnNumber - 1] = X;
             } else {
                 System.out.println(NOT_EMPTY_PLACE);
                 continue;
             }
-            drawField(boardSize, tab);
+            drawField(boardSize, board);
             //sprawdzenie
-            shouldPlayAgain = verifyIfContinue(boardSize, tab);
+            shouldPlayAgain = verifyIfContinue(boardSize, board);
 
             //ruch "O"
-            //drawField(boardSize,tab);
+            //drawField(boardSize,board);
 
 
         } while (shouldPlayAgain);
+        System.out.println(END_OF_THE_GAME);
     }
 
-    public static boolean verifyIfContinue(int boardSize, int[][] tab) {
-        if (verifyIfRowIsNotFull(boardSize, tab)) {
-            if (verifyIfColumnIsNotFull(boardSize, tab)) {
-                if (verifyIfDiagonalXxIsNotFull(boardSize, tab)) {
-                    if (verifyIfDiagonalYyIsNotFull(boardSize, tab)) {
-                        return true;
-                    }
-                }
-            }
-        }
-        System.out.println(END_OF_THE_GAME);
-        return false;
+    private static boolean verifyIfContinue(int boardSize, final PlayerSignType[][] board) {
+        return verifyIfRowIsNotFull(boardSize, board)
+                && verifyIfColumnIsNotFull(boardSize, board)
+                && verifyIfDiagonalXxIsNotFull(boardSize, board)
+                && verifyIfDiagonalYyIsNotFull(boardSize, board);
     }
 
-    public static boolean verifyIfDiagonalYyIsNotFull(int boardSize, int[][] tab) {
-        int[][] checking = new int[1][2];
-        checking[0][0] = 0;
-        checking[0][1] = 0;
+    public static boolean verifyIfDiagonalYyIsNotFull(int boardSize, final PlayerSignType[][] board) {
+        int intForCheck = 0;
 
-        for (int i = 0; i < tab.length; i++) {
-            if (tab[i][boardSize - i - 1] == ItIsX) checking[0][0] += 1;   // X
-            if (tab[i][boardSize - i - 1] == ItIsO) checking[0][1] += 1;   // O
+        for (int i = 0; i < board.length; i++) {
+            if (board[i][boardSize - i - 1] == PlayerSignType.X) intForCheck += 1;   // X
+            if (board[i][boardSize - i - 1] == PlayerSignType.O) intForCheck -= 1;   // O
```

Comment:
Powiedz mi jeszcze dlaczego tu są dwa ify zamiast `if() {} else if() {}` ?
Przez to oba warunki zawsze są sprawdzane mimo iż się wzajemnie wykluczają.

Reply:
Dlatego że to nie są jedyne opcje (może jeszcze być empty), więc za drugim razem nie mogę założyć że na pewno jest O, czyli gdybym zrobiła else to i tak muszę w środku zrobić jeszcze if. Czyli tak jest krócej.

Reply:
No nie, możesz zrobić:
```
if(coś) {
} else if (coś2) {
} else {
}
```
i wtedy w przypadku gdy pierwszy warunek będzie prawdą to reszta już nie będzie sprawdzana.
Swoją drogą to równie dobrze mógłby być switch.

Reply:
Zrobiła switch, nawet lepiej wygląda;)